components/Deck/InfoPanel/TranslationItem.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import {NavLink} from 'fluxible-router';
+import qs from 'querystring';
+
+import {getLanguageName} from '../../../common';
+import {flagForLocale} from '../../../configs/locales';
+
+class TranslationItem extends React.Component {
+    render() {
+        // build the href!
+        let routeName = 'deck';
+        let navParams = Object.assign({ slug: this.props.slug || '_' }, this.props.selector);
+        let queryParams = { language: this.props.language };
+
+        let href = ['', routeName, navParams.id, navParams.slug, navParams.stype, navParams.sid, navParams.spath].join('/');
```

Comment:
Why not using the Util component for URL building?

Reply:
I started out using some other way to do this that would require less data than what I ended up using, and then forgot about the Util method. I'm not sure if it would work either, as it seems i couldn't access window.location from the component code, at least not the render() method, which kinda makes sense

Reply:
Util.makeNodeURL(this.props.selector, 'deck', undefined, this.props.selector.slug, this.props.language) will do the job.
window.location is just used optionally at the beginning of the makeNodeURL method, at the end the URL is the same.